### ISSUE TITLE
Incorrectly reporting CA1822: Member can be marked static

### ISSUE THREAD

**Issue by Soreloser2** [dotnet/roslyn#63429]
**Version Used**: 17.4.0 P2 (32815.475.main)
I was able to repro both with and without LSP pull diagnostics enabled.

**Steps to Reproduce**:

1. Write code such as:
```
 public class C
    {
        private readonly int _value;

        public C(int foo)
        {
            _value = foo;
        }

        public void M()
        {
            Console.WriteLine(_value);
        }
    }
```
2. Hover over/notice the squiggle over "M"

**Expected Behavior**:
Do not see CA1822 as the method accesses the instance variable `_value`

**Actual Behavior**:
![image](https://user-images.githubusercontent.com/43689321/184955534-5a5fc7ae-4382-403d-835c-d159d59a16ce.png)
Note: the quick actions/fixes menu offers a "make member static" fix, which adds the static keyword, but this introduces an error as we now reference an instance field in a static method.

**Comment by Youssef1313** [comment]
It's so weird that CA1822 won't work for this simple case.

@mavasani Could this be some sort of Error List stuck diagnostics issue, ie, https://github.com/dotnet/roslyn/issues/51521 ?

If it's not, this needs to be moved to roslyn-analyzers for investigation.

**Comment by Soreloser2** [comment]
I don't think this is an issue with it sticking. The diagnostic will show up after re-opening the solution, or after adding `static`, waiting for the diagnostic to go away, then adding the `static` back in.

**Comment by Youssef1313** [comment]
![no-repro](https://user-images.githubusercontent.com/31348972/184981008-7aca4481-6c00-414d-8f49-1ddd19effb31.gif)

I'm not able to repro.

**Comment by Soreloser2** [comment]
Interesting. I also can't repro on 17.4.0 P1 or earlier. It might be a new change or just something is messed up with my install specifically.

**Comment by Youssef1313** [comment]
@Soreloser2 In case you encountered this again, I think it's interesting to know if you see the error in command-line builds (if you do, having a binlog might be useful).

**Comment by dibarbet** [comment]
duplicate - https://github.com/dotnet/roslyn/issues/63466
